cdap-data-fabric/src/main/java/io/cdap/cdap/data2/metadata/dataset/MetadataDataset.java

```diff
@@ -711,7 +712,10 @@ private SearchResults searchByDefaultIndex(SearchRequest request) {
         while ((next = scanner.next()) != null) {
           Optional<MetadataEntry> metadataEntry = parseRow(next, column, request.getTypes(),
                                                            request.shouldShowHidden());
-          metadataEntry.ifPresent(results::add);
+          metadataEntry.map(e -> new MetadataResultEntry(e, searchTerm.getTerm())).ifPresent(results::add);
+          if (metadataEntry.isPresent()) {
```

Comment:
We should remove the if statement right?

Reply:
yes, you're right